tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp

```diff
@@ -32,12 +31,11 @@ void test_definite_integral_1d(const Index<1>& index_1d) {
 }
 
 void test_definite_integral_2d(const Index<2>& index_2d) {
-  Mesh<2> extents(index_2d);
```

Comment:
@osheamonn updated the PR as I was about the add this comment to my first review:

I don't much like calling the function argument `index_2d`. This name provides no clue as to the argument's purpose (the object is not used for lookup or organization) and only serves to duplicate the information already encoded by the type. Calling the argument something like `mesh` would give the user a clue as to what concept this object represents, and (for me at least) this would make the code a bit easier to understand.

So I propose that instead of a `Mesh` type, we simply call mesh-like instances of `Index<N>` as `mesh`. I recognize that I have a bias towards SpEC naming when making this proposal... Thoughts?